pkg/controller/generic/transform/transform_test.go

```diff
@@ -537,6 +562,10 @@ func setup(t *testing.T, f func(ctx context.Context, st state.State, rt *runtime
 		if err != nil && !errors.Is(err, context.Canceled) {
 			t.Fatal(err)
 		}
+
+		if onShutdown != nil {
```

Comment:
you could attach this via `t.Cleanup()` probably in the test itself, no need to pass it through here